dbms/src/Flash/Mpp/MPPTaskManager.h

```diff
@@ -54,6 +53,35 @@ struct MPPQueryTaskSet
         return state == Normal || state == Aborted;
     }
     ~MPPQueryTaskSet();
+    MPPTask * findMPPTask(const MPPTaskId & task_id) const;
+    bool isTaskRegistered(const MPPTaskId & task_id) const
+    {
+        return task_map.find(task_id) != task_map.end();
+    }
+    void registerTask(const MPPTaskId & task_id)
```

Comment:
It will be great if we don't need to maintain three status for MPPTask: unregistered, registered, visible.

Reply:
There is actually two status for MPPTask, visible or not visible, register is just a internal state and a task can be unregistered mutiple times, so basically no status need to be maintained for register/unregister

Reply:
Yeah, I mean we can try avoid the internal "registered while not visible" state.

Reply:
OK, I think I can try to refine it later.

Reply:
all right~